pkg/utils/io.go

```diff
@@ -76,3 +77,29 @@ func WriteTempFile(name string, data []byte, perm os.FileMode) (string, error) {
 
 	return tmpFileName, nil
 }
+
+func Memfile(name string, b []byte) (int, error) {
+	fd, err := unix.MemfdCreate(name, 0)
+	if err != nil {
+		return 0, fmt.Errorf("MemfdCreate: %v", err)
+	}
+
+	err = unix.Ftruncate(fd, int64(len(b)))
+	if err != nil {
+		return 0, fmt.Errorf("Ftruncate: %v", err)
+	}
+
+	data, err := unix.Mmap(fd, 0, len(b), unix.PROT_READ|unix.PROT_WRITE, unix.MAP_SHARED)
+	if err != nil {
+		return 0, fmt.Errorf("Mmap: %v", err)
+	}
+
+	copy(data, b)
+
+	err = unix.Munmap(data)
+	if err != nil {
+		return 0, fmt.Errorf("Munmap: %v", err)
+	}
```

Comment:
I don't fully understand this pattern, in part because Go doesn't provide any docs on this command, and linux's [mmap](https://www.man7.org/linux/man-pages/man2/mmap.2.html) doesn't take an fd. How is that we can copy to `data` and then immediately unmap it from memory?

Reply:
Great question. The snippet came from [this blog post](https://terinstock.com/post/2018/10/memfd_create-Temporary-in-memory-files-with-Go-and-Linux/) via @zbuc. I also don't fully understand the pattern, but it does indeed work..